src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs

```diff
@@ -419,31 +532,38 @@ private async void ToggleSheetVisibility()
                     _ => throw new ArgumentOutOfRangeException()
                 };
 
-                
                 //Set position based on size of content
                 if (Position <= 0)
                 {
                     //Calculate what size the content needs if the position is set to 0
-                    Position = m_sheetView.SheetContentHeighRequest / m_modalityLayout.Height;
+                        Position = m_sheetView.SheetContentHeightRequest / m_modalityLayout.Height;
                 }
                 else //Set position from input
                 {
                     await TranslateBasedOnPosition(false);
                     OnOpenCommand?.Execute(OnOpenCommandParameter);
+                    OnOpen?.Invoke(this, new EventArgs());
                 }
             }
-            else { 
-                var y = Alignment switch { 
-                        AlignmentOptions.Bottom => m_modalityLayout.Height,
-                        AlignmentOptions.Top => -m_modalityLayout.Height,
-                        _ => throw new ArgumentOutOfRangeException()
-                    };
-                m_modalityLayout.Hide(m_sheetView.SheetFrame, m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y), () => OnCloseCommand?.Execute(OnCloseCommandParameter));
+            else
+            {
+                if (m_sheetView == null) return;

```

Comment:
Nullcheck? Didn't you just create the variable?

Reply:
No, I didn't. I create it in the `if` branch, and check for null and use it in the `else` branch,